boards/clue_nrf52840/src/main.rs

```diff
@@ -135,6 +135,13 @@ fn baud_rate_reset_bootloader_enter() {
     }
 }
 
+// Function for the process console to use to reboot the board
+fn reboot() {
+    unsafe {
+        cortexm4::scb::reset();
+    }
+}
+
```

Comment:
I'm unsure whether this should really be an unsafe function. It's difficult to reason about this from a Rust safety-perspective, given we're literally resetting all state. A valid argument for keeping it safe may be that `std::process::exit` is also a safe function.

In either case, it should return the never type though. We can guarantee to Rust that this function indeed never returns by simply adding a loop which will (assuming that the reset is indeed performed) just never be entered.

```suggestion
fn reboot() -> ! {
    unsafe {
        cortexm4::scb::reset();
    }
    loop {}
}
```

Reply:
This seems to be an unstable feature https://github.com/rust-lang/rust/issues/35121.

Reply:
> I'm unsure whether this should really be an unsafe function. It's difficult to reason about this from a Rust safety-perspective, given we're literally resetting all state. A valid argument for keeping it safe may be that `std::process::exit` is also a safe function.
> 
The `cortexm4::scb::reset` function is unsafe.

Reply:
Done, thank you @lschuermann.